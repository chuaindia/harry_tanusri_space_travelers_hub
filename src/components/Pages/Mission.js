import React from 'react';

function Mission() {
  return (
    <div>
      <h1>Mission&lsquo;s Page</h1>
    </div>
  );
}

export default Mission;