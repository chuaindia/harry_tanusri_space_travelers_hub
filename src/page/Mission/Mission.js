import React from 'react';
import MissionStructure from '../../components/MissionStructure/MissionStructure';
import styles from './Mission.module.css';

const Mission = () => {
  const missions = [
    {
      id: 1,
      name: 'Telstar',
      description: 'Telstar 18V was launched on a SpaceX Falcon 9 Block 5 rocket from Space Launch Complex 40 (SLC40) at Cape Canaveral Air Force Station, Space Coast, Florida, United States, on September 10, 2018, at 12:45 AM EDT (4:45 UTC). It was deployed into a subsynchronous transfer orbit.',
    },
    {
      id: 2,
      name: 'Telstar',
      description: 'Telstar 18V was launched on a SpaceX Falcon 9 Block 5 rocket from space Launch Complex 40 (SLC40) at Cape Canaveral Air Force Station, Space Coast, Florida, United States, on September 10, 2018, at 12:45 AM EDT (4:45 UTC). It was deployed into a subsynchronous transfer orbit.',
    },
    {
      id: 3,
      name: 'Telstar',
      description: 'Telstar 18V was launched on a SpaceX Falcon 9 Block 5 rocket from space Launch Complex 40 (SLC40) at Cape Canaveral Air Force Station, Space Coast, Florida, United States, on September 10, 2018, at 12:45 AM EDT (4:45 UTC). It was deployed into a subsynchronous transfer orbit.',
    },
  ];
  return (
    <table className={styles.missionWrapper}>
      <thead>
        <tr className={styles.missionRow}>
          <th className={styles.missionTable}>Mission</th>
          <th className={styles.missionTable}>Description</th>
          <th className={styles.missionTable}>Status</th>
        </tr>
      </thead>
      <tbody>
        {missions.map((item) => (
          <tr className={styles.missionRows} key={item.id}>
            <MissionStructure name={item.name} description={item.description} />
          </tr>
        ))}
      </tbody>
    </table>
  );
};
export default Mission;
