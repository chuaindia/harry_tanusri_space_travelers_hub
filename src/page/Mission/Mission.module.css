.missionWrapper {
    width: 90%;
    margin-top: 1rem;
  }
  
  .missionWrapper,
  .missionRow,
  .missionTable,
  .missionRows {
    border: 1px solid #e2e5e9;
    border-collapse: collapse;
  }
  
  .missionTable {
    padding: 0.8rem;
  }