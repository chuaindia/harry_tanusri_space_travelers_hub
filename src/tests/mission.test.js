/* eslint-disable max-len */
import React from 'react';
import { screen } from '@testing-library/react';
import '@testing-library/jest-dom';
// We're using our own custom render function and not RTL's render.
import renderWithProviders from '../utils/test.util';
import Mission from '../components/mission';

describe('test for mission component', () => {
  test('renders mission component', () => {
    expect(renderWithProviders(<Mission description="some text here" reserved={false} missionName="name" missionId="FA1" />)).toMatchSnapshot();
  });

  test('displays the right button conditionally using reserved prop', () => {
    renderWithProviders(<Mission description="description" reserved={false} missionName="Space" missionId="FAD123" />);

    // "Not a member" should be in the document while "Active member" should not be in the document because reserved is falsy
    expect(screen.queryByText(/Not a member/i)).not.toBeNull();
    expect(screen.queryByText(/Active member/i)).toBeNull();
  });

  it('should listen to number of button click', () => {
    renderWithProviders(<Mission description="description" reserved={false} missionName="Space" missionId="FAD123" />);

    expect(screen.queryByText(/Space/i)).not.toBeNull();
    expect(screen.queryByText(/description/i)).not.toBeNull();
  });
});
