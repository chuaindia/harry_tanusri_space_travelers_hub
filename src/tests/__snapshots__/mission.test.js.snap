// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test for mission component renders mission component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="ps-2 mission row"
      >
        <div
          class="col-3 column col"
        >
          <h4>
            name
          </h4>
        </div>
        <div
          class="col-6 column col"
        >
          <p>
            some text here
          </p>
        </div>
        <div
          class="d-flex align-items-center justify-content-center column col"
        >
          <div
            class="fade p-1 not_a_member alert alert-secondary show"
            role="alert"
          >
            Not a member
          </div>
        </div>
        <div
          class="d-flex align-items-center justify-content-center col"
        >
          <button
            class="join_btn btn btn-danger"
            type="button"
          >
            Join Mission
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="ps-2 mission row"
    >
      <div
        class="col-3 column col"
      >
        <h4>
          name
        </h4>
      </div>
      <div
        class="col-6 column col"
      >
        <p>
          some text here
        </p>
      </div>
      <div
        class="d-flex align-items-center justify-content-center column col"
      >
        <div
          class="fade p-1 not_a_member alert alert-secondary show"
          role="alert"
        >
          Not a member
        </div>
      </div>
      <div
        class="d-flex align-items-center justify-content-center col"
      >
        <button
          class="join_btn btn btn-danger"
          type="button"
        >
          Join Mission
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": Object {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
